### PR TITLE
App: Add mobile and tablet styles

### PR DESCRIPTION
**Pull request by Kharya1337**

### CODE REVIEW

**Review by MarikTar — CHANGES_REQUESTED** [review]
Screenshot 2021-03-26 at 14 23 07

Fix please

File: app/src/components/header/index.module.sass [hunk 1]
```diff
@@ -14,10 +16,49 @@
...
   display: block
+  &_mobile_img
+    display: none
 
 .spacer
   flex: 1 1 auto
 
 .nav_item
   padding: 0 10px
+  &:nth-child(4)
```
Comment (MarikTar) [inline app/src/components/header/index.module.sass:27]: Hide this element by class and not by order

File: app/src/components/header/index.module.sass [hunk 2]
```diff
@@ -14,10 +16,49 @@
...
+      vertical-align: middle
+      height: 30px
+      width: 24px
+
+  .spacer
+    order: 1
 
+  .nav_item
+    padding: 10px 5px
+    &:nth-child(3)
```
Comment (MarikTar) [inline app/src/components/header/index.module.sass:52]: Add class please

File: app/src/components/sidebar/sidebar.module.sass [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+.root
+  width: 350px
+  height: 100%
+  max-height: calc(100vh - 80px)
+  display: none
+  vertical-align: top
+  position: fixed
+  left: 0
+  top: 80px
+  &.visible
```
Comment (MarikTar) [inline app/src/components/sidebar/sidebar.module.sass:12]: Conflict of action with name

File: app/src/pages/example/index.tsx [hunk 4]
```diff
@@ -479,36 +502,56 @@ export default class ExamplePage
...
+                <span ref={this.refFps} />
+              </Typography>
+            </div>
+          )}
           <div className={s.content_wrapper}>
             <Box
-              borderRadius={8}
-              stroke="grey_2"
+              borderRadius={isMobile ? 0 : 8}
+              stroke={isMobile ? '' : 'grey_2'}
```
Comment (MarikTar) [inline app/src/pages/example/index.tsx:554]:
```
 borderRadius={!isMobile && 8}
stroke={!isMobile && 'grey_2'}
```

File: app/src/pages/example/params.module.sass [hunk 5]
```diff
@@ -50,3 +52,33 @@
...
+    top: 0
+    width: 100%
+    height: 100%
+    margin-left: 0
+
+  .controller_header
+    &_wrapper
+      padding: 12px 30px 16px
+    &_title
+      color: white
```
Comment (MarikTar) [inline app/src/pages/example/params.module.sass:69]: Use `rgb(var(--white))`, also and in other places

File: app/src/pages/example/params.tsx [hunk 6]
```diff
@@ -55,7 +63,7 @@ export default class ParamsWrapper extends React.Component<IParamsWrapperProps>
             </div>
             <Typography
               type="b3"
-              color="dark"
+              color={isMobile ? 'white' : 'dark'}
```
Comment (MarikTar) [inline app/src/pages/example/params.tsx:66]: Duplicated code, please moved to variable

File: app/src/pages/example/params.tsx [hunk 7]
```diff
@@ -164,7 +180,7 @@ export default class ParamsWrapper extends React.Component<IParamsWrapperProps>
               return (
                 <Box
                   key={paramName}
-                  stroke="grey_2"
+                  stroke={isMobile ? '' : 'grey_2'}
```
Comment (MarikTar) [inline app/src/pages/example/params.tsx:183]: `!isMobile && 'grey_2'`

File: app/src/pages/example/params.tsx [hunk 8]
```diff
@@ -104,7 +112,7 @@ export default class ParamsWrapper extends React.Component<IParamsWrapperProps>
             </div>
             <Typography
               type="h5"
-              color="dark_grey"
+              color={isMobile ? 'white' : 'dark_grey'}
```
Comment (MarikTar) [inline app/src/pages/example/params.tsx:115]: Duplicated code, please moved to variable

**Comment by MarikTar** [comment]
In `math_pixelwise` showing error and button `Try again` but need show `Try another examples`.

**Review by MarikTar — CHANGES_REQUESTED** [review]

File: app/src/components/sidebar/sidebar_group.module.sass [hunk 9]
```diff
@@ -18,4 +20,18 @@
...
   &:hover,
   &:focus
-    opacity: .6
\ No newline at end of file
+    opacity: .6
+
+@include tabletMobile()
+  .root
+    &:last-child
+      margin-bottom: 57px
```
Comment (MarikTar) [inline app/src/components/sidebar/sidebar_group.module.sass:28]: image

This not resolve the disappearance of the bottom padding when scrolling on firefox

File: app/src/pages/docs/index.module.sass [hunk 10]
```diff
@@ -71,10 +73,70 @@
...
+      display: grid
+      grid-template-columns: .35fr .65fr
+      grid-template-rows: auto 1fr
+      border-top: 1px solid #f6f6f6
+      padding: 21px 0 30px
+      gap: 0 10px
+      grid-template-areas: "first third" "second third"
+      align-items: start
+      &:last-child
+        border-bottom: 1px solid #f6f6f6
```
Comment (donskov) [inline app/src/pages/docs/index.module.sass:128]: Use color from the theme.

File: app/src/pages/docs/index.module.sass [hunk 11]
```diff
@@ -71,10 +73,70 @@
...
+      margin-bottom: 55px
+
+    h6
+      margin-bottom: 5px
+
+    tr
+      display: grid
+      grid-template-columns: .35fr .65fr
+      grid-template-rows: auto 1fr
+      border-top: 1px solid #f6f6f6
```
Comment (donskov) [inline app/src/pages/docs/index.module.sass:122]: Use color from the theme.

File: app/src/pages/example/index.module.sass [hunk 12]
```diff
@@ -75,26 +75,89 @@
...
+
+  .top_title
+    &_wrapper
+      position: absolute
+      top: 20px
+      left: 25px
+      margin-bottom: 0
+      z-index: 1
+    &_text
+      color: #f4f5f9
```
Comment (donskov) [inline app/src/pages/example/index.module.sass:169]: Use color from the theme.

File: app/src/pages/example/params.tsx [hunk 13]
```diff
@@ -120,20 +129,27 @@ export default class ParamsWrapper extends React.Component<IParamsWrapperProps>
...
     const { intl } = this.context;
 
     if (params) {
       const listParams = Object.keys(params);
 
       return (
         <Box
-          borderRadius={8}
-          stroke="grey_2"
+          borderRadius={isMobile ? 0 : 8}
```
Comment (donskov) [inline app/src/pages/example/params.tsx:145]: Please check border color on mobile.

File: app/pages/_app.tsx [hunk 14]
```diff
@@ -26,9 +52,17 @@ export default function MyApp({ Component, pageProps }) {
...
+
       <IntlWrapper messages={en}>
         <Header
           isMain={isMain}
+          displaySidebar={(state) => setShowSidebar(state)}
+          showSidebar={showSidebar}
+        />
+        <Sidebar
+          config={config}
+          visible={showSidebar}
```
Comment (donskov) [inline app/pages/_app.tsx:64]: Unclear how to use `visible` and `isDocs` props. Unclear how to add styles based on these props.

File: app/src/components/header/index.module.sass [hunk 15]
```diff
@@ -14,10 +14,54 @@
...
+    display: block
+    order: 1
+    padding: 0
+    border-style: none
+    img
+      vertical-align: middle
+      margin-top: -2px
+      margin-left: 10px
+      display: inline-block
+      object-fit: none
```
Comment (donskov) [inline app/src/components/header/index.module.sass:66]: I think this style is unused. Please check.

File: app/src/components/sidebar/sidebar.module.sass [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+
+@media (max-width: 1024px)
+  .root
+    width: 100%
+    left: 0
+    top: 60px
+    max-height: calc(100% - 60px)
+    z-index: 5
+    overflow: auto
+    &.docs_page
```
Comment (donskov) [inline app/src/components/sidebar/sidebar.module.sass:50]: Unclear styles using.

**Review by donskov — CHANGES_REQUESTED** [review]

File: app/src/pages/example/params.tsx (hunk 13 above)
Comment (Kharya1337) [reply app/src/pages/example/params.tsx:145]: On mobile is no border on this Box

**Pull request merged**